ajax/actorinformation.php

```diff
@@ -43,243 +43,74 @@
 
 Session::checkLoginUser();
 
+// save value and force boolval for security of $_REQUEST['only_number]
 $only_number = boolval($_REQUEST['only_number'] ?? false);
 
-if (isset($_REQUEST['users_id_requester']) && ($_REQUEST['users_id_requester'] > 0)) {
-    $ticket = new Ticket();
-
-    $options2 = [
-        'criteria' => [
-            [
-                'field'      => 4, // users_id_requester
-                'searchtype' => 'equals',
-                'value'      => $_REQUEST['users_id_requester'],
-                'link'       => 'AND',
-            ],
-            [
-                'field'      => 12, // status
-                'searchtype' => 'equals',
-                'value'      => 'notold',
-                'link'       => 'AND',
-            ],
-        ],
-        'reset' => 'reset',
-    ];
-
-    $url = $ticket->getSearchURL() . "?" . Toolbox::append_params($options2, '&amp;');
-    $nb  = $ticket->countActiveObjectsForUser($_REQUEST['users_id_requester']);
-
-    if ($only_number) {
-        if ($nb > 0) {
-            echo "<a href='$url'>" . $nb . "</a>";
-        }
-    } else {
-        echo "&nbsp;<a href='$url' title=\"" . __s('Processing') . "\">(";
-        printf(__('%1$s: %2$s'), __('Processing'), $nb);
-        echo ")</a>";
+// force intval for security of all others $_REQUEST
+$_REQUEST = array_map('intval', $_REQUEST);
+
+// check if user is allowed to see the item (only if not current connected user)
+if (
+    $_REQUEST[key($_REQUEST)] != Session::getLoginUserID()
+    && preg_match('/^(users|groups|suppliers)_id_(.*)$/', key($_REQUEST), $matches)
+) {
```

Comment:
`key` is using current array key. I guess that the current key may change over time if request parameters are not passed in same order, and I think it is not a good idea to force parameters order.

```php
$actor_keys = preg_grep('/^(users|groups|suppliers)_id_(.*)$/', array_keys($_REQUEST));
if (count($actor_keys) !== 1) {
    // Unexpected request
    return;
}

$actor_key = reset($actor_keys);
$actor_id    = (int)$_REQUEST[$actor_key];
```